### PR TITLE
add beans

### PR DESCRIPTION
**Pull request by Slavacheban**

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -2,7 +2,8 @@
 <project version="4">
```
Comment (Anderwerty) [inline .idea/workspace.xml]: This file should not be in the repository.

File: pom.xml [hunk 2]
```diff
@@ -8,5 +8,28 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.17</version>
+        </dependency>
+    </dependencies>
 
 </project>
```
Comment (Anderwerty) [inline pom.xml]: Please add a blank line at the end of the file.

File: src/main/java/Application.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+public class Application {
+    public static void main(String[] args) {
+        ApplicationContext context = new AnnotationConfigApplicationContext("F:\\Slava\\spring\\src\\main\\java\\beans");
+        FootballClub shakhtarDonetsk = context.getBean(ShakhtarDonetsk.class);
+        FootballClub shakhtarDonetsk2 = context.getBean(ShakhtarDonetsk.class);
+        FootballClub dinamoKyev = context.getBean(DinamoKyev.class);
+        FootballClub dinamoKyev2 = context.getBean(DinamoKyev.class);
+
+        System.out.println(shakhtarDonetsk == shakhtarDonetsk2);
```
Comment (Anderwerty) [inline src/main/java/Application.java]: Please don't use System.out.println()

File: src/main/java/Application.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import beans.DinamoKyev;
+import beans.FootballClub;
+import beans.ShakhtarDonetsk;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+public class Application {
+    public static void main(String[] args) {
+        ApplicationContext context = new AnnotationConfigApplicationContext("F:\\Slava\\spring\\src\\main\\java\\beans");
```
Comment (Anderwerty) [inline src/main/java/Application.java]: Please use a relative path.

File: src/main/java/beans/DinamoKyev.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import javax.annotation.PreDestroy;
+
+@Getter
+@Setter
+@Component
+@NoArgsConstructor
+@AllArgsConstructor
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class DinamoKyev implements FootballClub {
+    private static Logger LOGGER = Logger.getLogger(ShakhtarDonetsk.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/DinamoKyev.java]: You forget the final modifier.

File: src/main/java/beans/ShakhtarDonetsk.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@Getter
+@Setter
+@Component
+@NoArgsConstructor
+@AllArgsConstructor
+public class ShakhtarDonetsk implements FootballClub {
+    private static Logger LOGGER = Logger.getLogger(ShakhtarDonetsk.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/ShakhtarDonetsk.java]: The same

File: src/main/java/beans/DinamoKyev.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@Getter
+@Setter
+@Component
+@NoArgsConstructor
+@AllArgsConstructor
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class DinamoKyev implements FootballClub {
```
Comment (Anderwerty) [inline src/main/java/beans/DinamoKyev.java:29]: Please use DI by using a constructor.
https://www.baeldung.com/spring-injection-lombok

File: src/main/java/beans/ShakhtarDonetsk.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@Getter
+@Setter
+@Component
+@NoArgsConstructor
+@AllArgsConstructor
+public class ShakhtarDonetsk implements FootballClub {
```
Comment (Anderwerty) [inline src/main/java/beans/ShakhtarDonetsk.java:26]: Please use DI by using a constructor.
https://www.baeldung.com/spring-injection-lombok

File: src/main/java/beans/backs/Isma.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("isma")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Isma implements Back {
+    private static Logger LOGGER = Logger.getLogger(Isma.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/backs/Isma.java]: The same

File: src/main/java/beans/backs/Ramos.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("ramos")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Ramos implements Back {
+    private static Logger LOGGER = Logger.getLogger(Ramos.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/backs/Ramos.java]: The same

File: src/main/java/beans/backs/Srna.java [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("srna")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Srna implements Back {
+    private static Logger LOGGER = Logger.getLogger(Srna.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/backs/Srna.java]: The same

File: src/main/java/beans/forvards/Adriano.java [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("adriano")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Adriano implements Forvard {
+    private static Logger LOGGER = Logger.getLogger(Adriano.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/forvards/Adriano.java]: The same

File: src/main/java/beans/forvards/Facundo.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+import beans.backs.Isma;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("facundo")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Facundo implements Forvard {
+    private static Logger LOGGER = Logger.getLogger(Isma.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/forvards/Facundo.java]: The same

File: src/main/java/beans/goalkeepers/Boyko.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("boyko")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Boyko implements Goalkeeper {
+    private static Logger LOGGER = Logger.getLogger(Boyko.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/goalkeepers/Boyko.java]: The same

File: src/main/java/beans/goalkeepers/Piat.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("piat")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Piat implements Goalkeeper {
+    private static Logger LOGGER = Logger.getLogger(Piat.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/goalkeepers/Piat.java]: The same

File: src/main/java/beans/halfbacks/Ferna.java [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("ferna")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Ferna implements HalfBack {
+    private static Logger LOGGER = Logger.getLogger(Ferna.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/halfbacks/Ferna.java]: The same

File: src/main/java/beans/halfbacks/Maycon.java [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("maycon")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Maycon implements HalfBack {
+    private static Logger LOGGER = Logger.getLogger(Maycon.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/halfbacks/Maycon.java]: The same

File: src/main/java/beans/halfbacks/Silva.java [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("silva")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Silva implements HalfBack {
+    private static Logger LOGGER = Logger.getLogger(Silva.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/halfbacks/Silva.java]: The same

File: src/main/java/beans/halfbacks/Taison.java [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component("taison")
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Taison implements HalfBack {
+    private static Logger LOGGER = Logger.getLogger(Taison.class.getName());
```
Comment (Anderwerty) [inline src/main/java/beans/halfbacks/Taison.java]: The same

**Review by Anderwerty — CHANGES_REQUESTED** [review]

File: .idea/workspace.xml (hunk 1 above)
Comment (Slavacheban) [reply .idea/workspace.xml]: i dont understand what you mean?

Comment (Anderwerty) [reply .idea/workspace.xml]: Please delete this file from repository

File: pom.xml [hunk 20]
```diff
@@ -8,5 +8,28 @@
     <artifactId>springHW</artifactId>
```
Comment (Anderwerty) [inline pom.xml:8]: Please remove this file from the repository.

File: springHW.iml [hunk 21]
```diff
@@ -11,5 +11,7 @@
     </content>
```
Comment (Anderwerty) [inline springHW.iml]: Please remove this file from the repository.

File: src/main/java/Application.java [hunk 22]
```diff
@@ -0,0 +1,21 @@
+import beans.DinamoKyev;
+import beans.FootballClub;
+import beans.ShakhtarDonetsk;
+import org.apache.log4j.Logger;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+public class Application {
+    private static final Logger LOGGER = Logger.getLogger(Application.class.getName());
+    public static void main(String[] args) {
```
Comment (Anderwerty) [inline src/main/java/Application.java:12]: Add a blank line between lines 9 and 10.

File: src/main/java/Application.java [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+        ApplicationContext context = new AnnotationConfigApplicationContext("src/main/java/beans");
+        FootballClub shakhtarDonetsk = context.getBean(ShakhtarDonetsk.class);
+        FootballClub shakhtarDonetsk2 = context.getBean(ShakhtarDonetsk.class);
+        FootballClub dinamoKyev = context.getBean(DinamoKyev.class);
+        FootballClub dinamoKyev2 = context.getBean(DinamoKyev.class);
+
+        LOGGER.info("shakhtarDonetsk = shakhtarDonetsk2 : " + (shakhtarDonetsk == shakhtarDonetsk2));
+        LOGGER.info("dinamoKyev = dinamoKyev2 : " + (dinamoKyev == dinamoKyev2));
+    }
+
```
Comment (Anderwerty) [inline src/main/java/Application.java]: Remove a redundant blank line.

File: src/main/java/beans/DinamoKyev.java [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+@Getter
+@Setter
+@Component
+@NoArgsConstructor
+@AllArgsConstructor
+@RequiredArgsConstructor
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class DinamoKyev implements FootballClub {
+    private static final Logger LOGGER = Logger.getLogger(ShakhtarDonetsk.class.getName());
+    @Autowired
```
Comment (Anderwerty) [inline src/main/java/beans/DinamoKyev.java:32]: Add a blank line between lines 30 and 31.

File: src/main/java/beans/ShakhtarDonetsk.java [hunk 25]
```diff
@@ -0,0 +1,57 @@
...
+
+@Getter
+@Setter
+@Component
+@NoArgsConstructor
+@AllArgsConstructor
+@RequiredArgsConstructor
+public class ShakhtarDonetsk implements FootballClub {
+    private static final Logger LOGGER = Logger.getLogger(ShakhtarDonetsk.class.getName());
+    @Autowired
```
Comment (Anderwerty) [inline src/main/java/beans/ShakhtarDonetsk.java:29]: Add a blank line between lines 28 and 29.

**Review by Anderwerty — CHANGES_REQUESTED** [review]
Why everywhere scope prototype?

File: pom.xml [hunk 26]
```diff
@@ -8,5 +8,28 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-core</artifactId>
+            <version>5.1.6.RELEASE</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>3.0.5.RELEASE</version>
```
Comment (Anderwerty) [inline pom.xml:20]: Why not version 5.1.*?
image

**Review by Anderwerty — APPROVED** [review]

**Pull request merged**